src/main/java/games/dominion/actions/Library.java

```diff
@@ -0,0 +1,93 @@
+package games.dominion.actions;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import core.AbstractGameState;
+import core.actions.AbstractAction;
+import core.actions.DoNothing;
+import core.interfaces.IExtendedSequence;
+import games.dominion.DominionGameState;
+import games.dominion.DominionConstants.DeckType;
+import games.dominion.cards.CardType;
+import games.dominion.cards.DominionCard;
+
+/**
+ * Action to play the Library card. This card allows the player to draw until they have 7 cards in hand, allowing them to discard any action cards drawn in the process.
+ */
+public class Library extends DominionAction implements IExtendedSequence {
+
+    boolean executed = false;
+
+    public Library(int playerId) {
+        super(CardType.LIBRARY, playerId);
+    }
+
+    public Library(int playerId, boolean dummy) {
+        super(CardType.LIBRARY, playerId, dummy);
+    }
+
+    @Override
+    public List<AbstractAction> _computeAvailableActions(AbstractGameState state) {
+        DominionGameState dgs = (DominionGameState) state;
+        List<AbstractAction> actions = new ArrayList<>();
+
+        // Make sure player can do this action
+        if (dgs.getDeck(DeckType.DRAW, player).getSize() == 0 || dgs.getDeck(DeckType.HAND, player).getSize() >= 7) {
+            actions.add(new DoNothing());
+            return actions;
+        }
```

Comment:
This is not needed (and is in fact wrong). If a player tries to draw a card when their draw deck is empty, then they shuffle their discard to become their new draw deck. This is all handled automatically within DominionGameState.drawCard(), so no need to do any special processing in individual actions.